databricks/koalas/indexing.py

```diff
@@ -377,12 +380,88 @@ def __setitem__(self, key, value):
         else:
             assert self._is_df
 
-            # TODO: support DataFrame.
-            raise SparkPandasNotImplementedError(
-                description="Assignment to DataFrame with the iloc indexer is not supported yet",
-                pandas_function=".{}[..., ...] = ...".format(type(self).__name__[:-7].lower()),
-                spark_target_function="withColumn, select",
-            )
+            if isinstance(key, tuple):
+                if len(key) != 2:
+                    raise SparkPandasIndexingError("Only accepts pairs of candidates")
+                rows_sel, cols_sel = key
+            else:
+                rows_sel = key
+                cols_sel = None
+
+            if isinstance(value, DataFrame):
+                if len(value.columns) == 1:
+                    value = _col(value)
+                else:
+                    raise ValueError("Only a dataframe with one column can be assigned")
+
+            if (isinstance(rows_sel, Series) and rows_sel._kdf is not self._kdf_or_kser) or (
+                isinstance(value, Series) and value._kdf is not self._kdf_or_kser
+            ):
+                kdf = self._kdf_or_kser.copy()
+                kdf["__temp_natural_order__"] = F.monotonically_increasing_id()
```

Comment:
Should we maybe reuse `__natural_order__` we defined in Koalas DataFrames?

Reply:
Unfortunately we can't since it will change after the assignment of key or value.